draft-ietf-quic-transport.md

```diff
@@ -2675,201 +2676,289 @@ created, it MUST close the connection with error code STREAM_STATE_ERROR.
-different offset from the one already established.
+After the application indicates that stream data is complete and a STREAM frame
+containing the FIN bit is sent, the send stream enters the "Data Sent" state.
+From this state, the endpoint only retransmits stream data as necessary.  The
+endpoint no longer needs to track flow control limits or send STREAM_BLOCKED
+frames for a send stream in this state.  The endpoint can ignore any
+MAX_STREAM_DATA frames it receives from its peer in this state; MAX_STREAM_DATA
+frames might be received until the peer receives the final stream offset.
 
-An endpoint will know the final offset of the data it receives on a stream when
-it reaches the "half-closed (remote)" state, see {{final-offset}} for details.
+Once all stream data has been successfully acknowledged, the send stream enters
+the "Data Recvd" state, which is a terminal state.
 
-A stream in this state can be used by the endpoint to send any frame that
-mentions a stream ID.  In this state, the endpoint MUST observe advertised
-stream and connection data limits (see {{flow-control}}).
+From any of the "Open", "Send", or "Data Sent" states, an application can signal
+that it wishes to abandon transmission of stream data.  Similarly, the endpoint
+might receive a STOP_SENDING frame from its peer.  This causes the endpoint to
+send a RST_STREAM frame, which causes the stream to enter the "Reset Sent"
+state.
 
-A stream transitions from this state to "closed" by completing transmission of
-all data.  This includes sending all data carried in STREAM frames including
-the terminal STREAM frame that contains a FIN flag.
+An endpoint MAY send a RST_STREAM to open a send stream; this causes the send
+stream to open and immediately transition to the "Reset Sent" state.
 
-A stream also becomes "closed" when the endpoint sends a RST_STREAM frame.
+Once a packet containing a RST_STREAM has been successfully acknowledged, the
+send stream enters the "Reset Recvd" state, which is a terminal state.
 
 
-### closed {#state-closed}
+### Receive Stream States {#stream-recv-states}
 
-The "closed" state is the terminal state for a stream.  Reordering might cause
-frames to be received after closing, see {{state-hc-remote}}.
+{{fig-stream-recv-states}} shows the states for the part of a stream that
+receives data from a peer.  The states for a receive stream mirror only some the
+states of the send stream at the peer.  A receive stream doesn't track states on
+the send stream that can't be observed, such as the "Open" state; instead,
+receive streams track the delivery of data to the application or application
+protocol.
 
-If the application resets a stream that is already in the "closed" state, a
-RST_STREAM frame MAY still be sent in order to cancel retransmissions of
-previously-sent STREAM frames.
+~~~
+       o
+       | Recv STREAM / STREAM_BLOCKED / RST_STREAM
+       | Open Paired Stream (bidirectional)
+       | Recv MAX_STREAM_DATA
+       v
```

Comment:
On the send side, you have separate states for "I know the stream is open" and "data is flowing."  On the receive side, you go immediately to the "Recv" state.  Is there value in splitting those?

Reply:
I don't think so?